### ISSUE TITLE
invalid code generated for preact useRef

### ISSUE THREAD

**Issue by nitsky** [evanw/esbuild#302]
I am observing that esbuild is emitting the invalid code shown below. `h3` corresponds to `useRef` and `o3` corresponds to `currentHook` from `preact/hooks`: https://github.com/preactjs/preact/blob/3017d91a69f1b52bb89be118e22f4778cb82dd02/hooks/src/index.js#L175. It seems esbuild is choosing to split off `currentHook` into a separate chunk even though its value is reassigned in `useRef`. I am using esbuild v0.6.11 and the `bundle` and `splitting` flags.

```
import {
  o3,
} from "./chunk.frH4ZD-M.js";

// ../node_modules/preact/hooks/dist/hooks.module.js
function h3(n3) {
  return o3 = 5, _3(function() {
    return {current: n3};
  }, []);
}
```

**Comment by evanw** [comment]
Thank you so much for reporting this. This is a really good test case. I've fixed this issue and the fix will be in the next release. I wrote up some more information about this bug [in the release notes](https://github.com/evanw/esbuild/blob/392680e812bbe5472f38073f4f49a3539fecfb07/CHANGELOG.md#unreleased).

**Comment by evanw** [comment]
This fix was released in version 0.6.12.

**Comment by nitsky** [comment]
@evanw thank you so much for the quick fix!
